### PR TITLE
Adds more documentation to existing tests.

### LINKED ISSUE #213: Document and adjust existing ocean.py tests
To document:
- [x] test_ocean_assets.py
- [x] test_data_service_provider.py
- [x] test_balancer_core_direct_flow.py
- [x] test_balancer_simple_for_users_flow.py
- [x] test_bfactory.py
- [ ] test_bpool.py -> almost done, uncertainties with join/exit in/out combinations for pools
- [x] test_btoken.py
- [x] test_datatoken.py
- [x] test_ddo.py
- [x] test_dtfactory.py
- [x] test_fixed_rate_exchange.py
- [x] test_ocean_auth.py
- [x] test_ocean_exchange.py
- [x] test_ocean_services.py
- [x] test_config_provider.py
- [x] test_contract_base.py
- [x] test_contract_handler.py -> split in separate files for abi_paths and addresses
- [x] test_wallet.py
- [x] test_compute_flow.py
- [x] test_market_flow.py

To adjust:
- [ ] in test_ocean_assets.py, test_register_asset needs to be broken down into separate tests. The existing one is testing too many paths and is hard to follow.
- [ ] in test_bpool.py, can we add some unsuccessful transfers in test_public_pool?
- [ ] add more tests to test_btoken.py
- [ ] in test_ddo, the test_ddo_on_chain test is much too complex, should be broken down. Potentially more tests are needed.
- [ ] need to add some sad flows for test_bfactory and test_dtfactory
- [ ] in test_fixed_rate_exchange the namesake test is long and too complex
- [ ] test_ocean_exchange.py, namesake test: too long and complex, remove stray prints and handle TODOs
- [ ] test_ocean_services.py: only creation is tested, need to add more tests regarding services.
- [ ] test_contract_base.py: FIXME at the end of the file + more tests.
- [ ] test_wallet.py needs more testing, only attribute setup is covered now

### PR DESCRIPTION
**Pull request by calina-c**
Fixes #213.

Changes proposed in this PR:

- document and review existing ocean.py tests

### CODE REVIEW

File: ocean_lib/ocean/test/test_ocean_auth.py [hunk 1]
```diff
@@ -31,9 +33,11 @@ def test_check_token(web3_instance):
     assert ocn_auth.check(sig) == "0x0"
 
     # Test token expiration
+    # TODO
```
Comment (trentmc) [inline ocean_lib/ocean/test/test_ocean_auth.py:36]: Was this meant to be completed in this ticket?

**Review by trentmc — APPROVED** [review]
All good, except there's one new "TODO" worth resolving. I'll "approve" anyway

File: ocean_lib/ocean/test/test_ocean_auth.py (hunk 1 above)
Comment (calina-c) [reply ocean_lib/ocean/test/test_ocean_auth.py:36]: no, I just replaced a FIXME with a TODO. I've also added this to the list of future improvements to the tests.
